controllers/clustermesh/clustermesh_controller.go

```diff
@@ -269,13 +266,23 @@ func (r *ClusterMeshReconciler) reconcileDelete(ctx context.Context, cluster *cl
 	}
 
 	err := r.Delete(ctx, sg)
-	if err != nil {
+	if err != nil && !apierrors.IsNotFound(err) {
 		return err
 	}
 	r.log.Info(fmt.Sprintf("deleted security group for cluster %s\n", cluster.ObjectMeta.Name))
 
-	if err := r.ReconcilePeeringsFactory(r, ctx, clustermesh); err != nil {
-		return err
+	clustermeshCopy := clustermesh.DeepCopy()
+
+	for _, vpcPeeringConnectionRef := range clustermeshCopy.Status.CrossplanePeeringRef {
```

Comment:
We could also just check if the cluster.Name is a substring of the CrossplanePeeringRef.Name WDYT?